File: mall-user-interface/src/main/java/com/mall/user/service/UserService.java
package com.mall.user.service;

import com.mall.pojo.User;
import common.pojo.PageResult;

import java.util.List;

/**
 * 用户服务接口
 * @author Wwl
 *
 */
public interface UserService {

	/**
	 * 增加
	 * @param user
	 */
	public void add(User user);

	/**
	 * 修改
	 * @param user
	 */
	public void update(User user);

	/**
	 * 根据Id获取实体
	 * @param id
	 * */
	public User findOne(Long id);

	/**
	 * 查询所有
	 * @return
	 */
	public List<User> findAll();

	/**
	 * 查询所有-分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 根据条件查询，分页
	 * @param user
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(User user, int pageNum, int pageSize);

	/**
	 * 删除
	 * @return
	 */
	public void delete(Long[] ids);

	/**
	 * 根据用户名称查询用户
	 * @param username
	 * @return
	 */
	public User findByUsername(String username);

	/**
	 * 生成短信验证码
	 * @param phone
	 */
	public void createSmsCode(String phone);

	/**
	 * 校验短信验证码
	 * @param phone
	 */
	public boolean checkSmsCode(String phone,String code);

}

File: mall-user-web/src/main/webapp/js/service/userService.js
/**
 * 用户服务层
 * Created by Wwl on 2019/3/4
 */
app.service('userService', function($http){

    //增加
    this.add = (entity,code) => $http.post('../user/add.do?code='+code,entity);

    //修改
    this.update = (entity) => $http.post('../user/update.do',entity);

    //删除
    this.delete = (ids) => $http.get('../user/delete.do?ids='+ids);

    //查询实体
    this.findOne = (id) => $http.get('../user/findOne.do?id='+id);

    //查询列表
    this.findList = () => $http.get('../user/list.do');

    //分页查询
    this.findPage = (page,size) => $http.get('../user/listByPage.do?page=' + page + '&size=' + size);

    //搜索
    this.search = (page,size,searchEntity) => $http.get('../user/search.do?page='+page+'&size='+size,searchEntity);

    //发送验证码
    this.sendCode = (phone) => $http.get('../user/sendCode.do?phone='+phone);


});